.azure-pipelines.yml

```diff
@@ -455,24 +460,24 @@ jobs:
       modifyEnvironment: true
   - bash: |
       set -eux
-      pip install pip --user --upgrade
+      # pip 20.3 introduces the new dependency resolver
+      pip install pip>=20.3 --user --upgrade
       # Path with backslash are a pain...
       SOURCES_DIRECTORY=`python -c 'print(r"$(Build.SourcesDirectory)".replace("\\\\", "/"))'`
       TEMP_DIRECTORY=`python -c 'print(r"$(Agent.TempDirectory)".replace("\\\\", "/"))'`
       VERBOSE_TAG=`git -C $SOURCES_DIRECTORY describe --tag`
-      sed -i "s#^\\(__version__ = \\).*\$#\\1'$VERBOSE_TAG'#" $SOURCES_DIRECTORY/parsec/_version.py
+      # sed -i "s#^\\(__version__ = \\).*\$#\\1'$VERBOSE_TAG'#" $SOURCES_DIRECTORY/parsec/_version.py
```

Comment:
yes, this is a bit shameful by the way:

The idea of this sed command is to customize the `__version__` attribute in the parsec module so that we know precisely on what code the module has been created.

However this `__version__` is also used in `setup.py` which is used to generate the wheel here.

But wheel generation requires a package version that respect [PEP440](https://www.python.org/dev/peps/pep-0440/) formatting, which is not the case for something like `v2.2.4-89-g82575bdd` :'(

The funny thing is this is not an issue when running the CI against a tag (given `git describe --tag` will then return the tag name, which is the same value that is already in `parsec/_version.py`'s `__version__` field (so the sed command doesn't change anything in this case).
BUT given this job is only run when we build a tag... it only fails when we are manually re-enabling for quick test like this PR is doing 😄 

So in the end what we can do:
1) nothing (uncomment the sed command), we consider the sed command act as a sanity check that must be manually overwritten when needed
2) remove the sed command entirely. Considering we basically never use the generated package (or only for very limited test, with no distribution of the package to anyone), we don't really need it anyway
3) fix the issue by separating `setup.py`'s version from `__version__`: for instance only applying the sed command to `__version__`, considering the `setup.py` value (e.g. `v2.4.2+dev`) is good enough

I would vote for 3 ;-)

Reply:
Why not use [local version identifiers](https://www.python.org/dev/peps/pep-0440/#local-version-identifiers), e.g `2.2.4.dev0+<commit hash>`?

Reply:
good idea, let's try this ^^
see bac38f3224b19b3b65db9c00dace0e1667de52e7